### ISSUE TITLE
Genome-wide recombination rate lowered by rate 0 chromosomes

### ISSUE THREAD

**Issue by jeromekelleher** [popsim-consortium/stdpopsim#799]
When we create a "generic" config of a given length for a species, we compute the recombination and mutation rates as the length weighted recombination rate over the whole genome, [here](https://github.com/popsim-consortium/stdpopsim/blob/main/stdpopsim/species.py#L206). We do exclude non-autosomes, but the detection is weak and won't work for most species.

Perhaps it would be simpler to just exclude chrs with zero recombination or mutation rate from the computation?

**Comment by grahamgower** [comment]
Good idea. What about also excluding regions of rate 0 (e.g. centromeres)?

**Comment by andrewkern** [comment]
but what would we do with a chromosome, like chr4 is drosophila that never recombines? we usually include that "0" when averaging

**Comment by grahamgower** [comment]
> but what would we do with a chromosome, like chr4 is drosophila that never recombines? we usually include that "0" when averaging

Interesting! I feel it would be a mistake to simulate using a value averaged from both recombining and non-recombining chromosomes though. Surely the user is better served by simulating something that either recombines (using a rate averaged over recombining regions), or simulating something that doesn't recombine?

**Comment by jeromekelleher** [comment]
Yes, that was my take as well @grahamgower.
